crates/matrix-sdk/src/send_queue.rs

```diff
@@ -1023,10 +1031,11 @@ impl QueueStorage {
         // Keep the lock until we're done touching the storage.
         let mut being_sent = self.being_sent.write().await;
         let was_being_sent = being_sent.take();
-        assert_eq!(
-            was_being_sent.as_ref().map(|info| info.transaction_id.as_ref()),
-            Some(transaction_id)
-        );
+
+        let prev_txn = was_being_sent.as_ref().map(|info| info.transaction_id.as_ref());
+        if prev_txn != Some(transaction_id) {
```

Comment:
The assert would have aborted execution here, while we now continue on, is that fine? Don't we want to perhaps return early?

Reply:
I think it would be a sign that the system is in a weird state, but the transaction_id passed as a parameter is the important value that is correct above all, so the current code is fine.